src/com/esotericsoftware/kryo/serializers/CompatibleFieldSerializer.java

```diff
@@ -64,11 +65,18 @@ public void write (Kryo kryo, Output output, T object) {
 
 		CachedField[] fields = cachedFields.fields;
 		ObjectMap context = kryo.getGraphContext();
+		ArrayList fieldsList = new ArrayList();
 		if (!context.containsKey(this)) {
 			if (TRACE) trace("kryo", "Write fields for class: " + type.getName());
 			context.put(this, null);
 			output.writeVarInt(fields.length, true);
 			for (int i = 0, n = fields.length; i < n; i++) {
+				fieldsList.add(fields[i].name);
+				if (i + 1 < n) {
+					if (fieldsList.contains(fields[i+1].name))

```

Comment:
This is quite inefficient. Why not use a `HashSet`? `HashSet.add()` returns `false` if the value is already present, so we can do the check in a single operation.